Add shellcheck workflow to lint shell scripts

emersonknapp opened the pull request:
Signed-off-by: Emerson Knapp <emerson.b.knapp@gmail.com>

thomas-moulard commented:
Nice - could you demonstrate that this actually fail if you have a bug in a shell script?

emersonknapp commented:
It's there in the commit history - https://github.com/ros-tooling/cross_compile/pull/148/checks?check_run_id=451846076 - you can see once I fixed the workflow syntax, the check failed because of a shellcheck complaint in `build_workspace.sh`, which I fixed in the next commit and it's green now.
